mono/metadata/icall.c

```diff
@@ -984,8 +984,12 @@ ves_icall_System_Runtime_CompilerServices_RuntimeHelpers_SufficientExecutionStac
 	char *end = info->stack_end;
 	char *current = (char*)&end;
 
-	/* min_size will be > 0 as sgen can't work otherwise */
-	size_t min_size = (end - start) / 2;
+	//This is a lot less than CoreCLR demands. We run with significantly smaller stacks.
+#if SIZEOF_VOID_P == 8
```

Comment:
I would suggest `min_size = SIZEOF_VOID_P * 8 * 1024` (8 kilopointers?) although it’d be 0 if `SIZEOF_VOID_P` were ever undefined.